backend/baboon.go

```
@@ -0,0 +1,238 @@
+		fmt.Println(baboonEndpoint)
+		u, err := url.Parse(baboonEndpoint)
+		if err != nil {
+			glog.Errorf("unable to parse rawurl, reason %s", err)
+		}
+		glog.Infof("about to remove F5 pool entity with AppID '%s' via calling '%s'", e.Appid, baboonEndpoint)
+
+		response, err = be.session.Delete(u.String(), nil, nil, nil)
+		if err != nil {
+			glog.Errorf("unable to remove pool '%s'", pool)
+		}
+		glog.Infof("DELETE response (%d): %s", response.Status(), response.RawText())
+	}
+}
+
+// createPoolMember calls baboon
+func (be Baboon) createPool(e ApiRequestEvent) {
+	var (
+		response *napping.Response
+	)
+
+	pool := strings.TrimLeft(e.Appdefinition.ID, "/")
+	dcs := strings.Split(be.datacenter, ",")
+	token := be.getToken()
+	be.session.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	payload := addPool{Name: pool, Monitor: "tcp"}
+	for i := range dcs {
+		// needs to be more general -ltm is hardcoded which is not so great
+		baboonEndpoint := fmt.Sprintf("%s%s-ltm/pools", be.f5PoolService, dcs[i])
+		fmt.Println(baboonEndpoint)
+		u, err := url.Parse(baboonEndpoint)
+		if err != nil {
+			glog.Errorf("unable to parse rawurl, reason %s", err)
+		}
+		glog.Infof("about to add F5 pool entity with AppID '%s' via calling '%s'", e.Appdefinition.ID, baboonEndpoint)
+
+		response, err = be.session.Post(u.String(), payload, nil, nil)
+		if err != nil {
+			glog.Errorf("unable to add pool '%s'", pool)
+		}
+		glog.Infof("POST response (%d): %s", response.Status(), response.RawText())
+	}
+}
+
+// modifyPoolMember calls baboon
+func (be Baboon) modifyPoolMember(e StatusUpdateEvent) {
+	var (
+		response *napping.Response
+		entity   *BaboonService
+	)
+
+	switch {
+	case e.Taskstatus == "TASK_RUNNING":
+		entity = &BaboonService{Type: "Add pool member"}
+	case e.Taskstatus == "TASK_KILLED":
+		entity = &BaboonService{Type: "Delete pool member"}
+	default:
+		entity = &BaboonService{Type: "Unknown type"}
```

Can we have a fail fast approach? Return/do nothing in case the task is unknown?